### PR TITLE
Feat/design system site

### PR DESCRIPTION
**Pull request by MildTomato**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

- init the Design System site

## What is the current behavior?

- no site

## What is the new behavior?

- has site

### CODE REVIEW

**Comment by MildTomato** [comment]
Screenshot 2024-05-21 at 4 25 45 PM

ignore this. we've exceeded number of file changes

**Review by ivasilov — APPROVED** [review]

**Review by alaister — APPROVED** [review]

File: apps/design-system/app/Providers.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+'use client'
+
+import * as React from 'react'
+import { Provider as JotaiProvider } from 'jotai'
```
Comment (saltcod) [inline apps/design-system/app/Providers.tsx:4]: curious why not valtio?

Comment (MildTomato) [reply apps/design-system/app/Providers.tsx:4]: Not sure it matters.
Large bits of the site was from ui.shadcn and they used jotai.

[This tweet](https://x.com/dai_shi/status/1343694177884721152) sums up the differences 😂

Comment (saltcod) [reply apps/design-system/app/Providers.tsx:4]: aah got it — kk just curious

**Pull request merged**